common/templates/structs.go

```diff
@@ -25,6 +26,8 @@ type CtxChannel struct {
 	PermissionOverwrites []*discordgo.PermissionOverwrite `json:"permission_overwrites"`
 	ParentID             int64                            `json:"parent_id"`
 	OwnerID              int64                            `json:"owner_id"`
+	AppliedTags          []string                         `json:"applied_tags"`

```

Comment:
Why is this a `[]string` when the Discord docs [^1] indicate an array
of snowflakes, i.e. `[]int64`?

Any reason not to have this as a `[]ForumTag`?
Obviously we'll be still ignoring the Discord docs for this, but a
`[]string` is really, really odd.

[^1]: https://discord.com/developers/docs/resources/channel#channel-object-channel-structure

Reply:
The object is `[]string` to allow compatibility within the `setForumTags` template
Would it make more sense to have it as `[]ForumTag`? How come if so?

Reply:
> The object is `[]string` to allow compatibility within the 
> `setForumTags` template

Nowhere in `tmplSetForumTags` are you returning a non-empty string.
Consider just returning the error instead, or having sensible return 
values and/or types.

>  Would it make more sense to have it as `[]ForumTag`? How come if so?

If it is of type `[]ForumTag` users would be able to directly `index`
said slice and have the `ForumTag` object ready from the get-go:

```go
{{/* Grab the first ForumTag and output its information */}}
{{ index .Channel.AppliedTags 0 | json }}
```

Otherwise, they would have to somehow get the `ForumTag` object, which
I do not see a template for in your patchset. Though, maybe also add 
such a function as well.

Reply:
Spent some time messing around with the AppliedTags object.
Figured out that for some unknown reason, despite the discord developer docs stating that it is an array of snowflakes, returning it in `[]int64` would result in a response of 0 for any tag.
Can't tell you why it's doing that so that's why I opted with `[]string`.

As for returning it with `[]ForumTag`, similar issue. We get a response and it returns data however it returns blank or again with 0's.

Will resolve the other conversations pertaining to this aside from this one. Waiting on a response from Des in programming discussion and we'll go from there

Reply:
Is this still pending?

Reply:
Yep